src/core/default/mappers/aws_client.py

```diff
@@ -1,13 +1,13 @@
 from time import sleep
-from typing import Callable, List
+from typing import Callable, List, Optional, Any
 import boto3
 
 AVAILABLE_SERVICES = set(
     ["lambda", "s3", "dynamodb", "iam", "apigatewayv2", "sqs", "apigateway"]
 )
 
 
-def _get_boto_client(service_name, credentials=None, profile_name=None):
+def _get_boto_client(service_name, credentials=None, profile_name=None) -> Optional[boto3.session.Session]:
```

Comment:
Lets not use Optional[] here and instead throw an exception. I think it will be a better practice to start being more explicit about throwing exceptions instead of just returning None. Especially in this case because not being to create a valid session will cause lots of downstream problems.

Reply:
In this case I think the logic is wrong:
```
    if credentials:
        return boto3.Session(
            aws_access_key_id=credentials.get("access_key"),
            aws_secret_access_key=credentials.get("secret_key"),
        ).client(service_name)

    if profile_name:
        return boto3.Session(profile_name=profile_name).client(service_name)

      return boto3.client(service_name)

```